### PR TITLE
Refactor alignment

### PR DESCRIPTION
**Pull request by beatngu13**
I will update this draft while I try to better understand the `Alignment` class. Please let me know if there is something you wouldn't change/refactor.

### CODE REVIEW

**Comment by beatngu13** [comment]
Oh and you better go commit by commit, the overall diff is awful.

**Comment by beatngu13** [comment]
[nudge reviewer]

**Review by roesslerj — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/ui/diff/Alignment.java [hunk 1]
```diff
@@ -64,12 +67,12 @@ private Alignment( final Element expected, final Element actual ) {
 	private Map<Element, Element> createAlignment( final List<Element> expectedElements,
 			final Map<Element, Element> actualElements ) {
 		final Map<Element, Match> matches = new HashMap<>();
-		final Stack<Element> elementsToAlign = toStack( expectedElements );
+		final Queue<Element> elementsToAlign = toReverseQueue( expectedElements );
```
[the remark this answers is not in the document]
Comment (beatngu13) [reply src/main/java/de/retest/recheck/ui/diff/Alignment.java]: I added this reference to the commit message:

https://stackoverflow.com/a/30389890

Makes sense to me, what do you say?

[the remark this answers is not in the document]
Comment (beatngu13) [reply src/main/java/de/retest/recheck/ui/diff/Alignment.java]: Oh and we reverse the stack too, it just wasn't reflected in the method name: https://github.com/retest/recheck/blob/master/src/main/java/de/retest/recheck/ui/diff/Alignment.java#L168

[the remark this answers is not in the document]
Comment (beatngu13) [reply src/main/java/de/retest/recheck/ui/diff/Alignment.java]: (Just realized I should use `Deque` …)

[the remark this answers is not in the document]
Comment (roesslerj) [reply src/main/java/de/retest/recheck/ui/diff/Alignment.java]: Wanna push a Dequeue? Then I'll approve :-)

**Review by roesslerj — APPROVED** [review]

**Pull request merged**